databases/postgresql16-dev/Portfile

```diff
@@ -0,0 +1,180 @@
+master_sites        postgresql:source/v${version}/
+distname            postgresql-${version}
+
+checksums           rmd160  e0ef5a9d1c7bf359a33f01b86e9f38abe30db20c \
+                    sha256  ffcf44e272662f6ac451a8d6d6ff951715db651c8d4907ec659cbde46abd52d3 \
+                    size    24313365
+
+use_bzip2           yes
+
+# do not build man or html files (use postgresqlXY-doc instead)
+patchfiles-append   patch-no_doc.diff
+
+# https://trac.macports.org/ticket/66060
+# https://trac.macports.org/ticket/67365
+# https://postgrespro.ru/list/thread-id/2551610
+# The problem is specific to PPC: https://github.com/macports/macports-ports/pull/16460
+platform darwin powerpc {
+    patchfiles-append  patch-icu.diff \
+                       patch-fix-ppc-asm.diff
+}
+
+depends_lib         port:readline path:lib/libssl.dylib:openssl port:zlib \
+                    port:libxml2 port:libxslt \
+                    path:lib/pkgconfig/icu-uc.pc:icu \
+                    path:lib/libgssapi_krb5.dylib:kerberos5 \
+                    path:lib/libzstd.dylib:zstd \
+                    path:lib/liblz4.dylib:lz4
+depends_build       port:bison port:pkgconfig
+depends_run         port:postgresql_select
+
+legacysupport.newest_darwin_requires_legacy \
+                    15
+worksrcdir          postgresql-${version}
+
+set libdir          ${prefix}/lib/${name}
+configure.args      PG_SYSROOT=disabled \
+                    --sysconfdir=${prefix}/etc/${name} \
+                    --bindir=${libdir}/bin \
+                    --libdir=${libdir} \
+                    --includedir=${prefix}/include/${name} \
+                    --datadir=${prefix}/share/${name} \
+                    --docdir=${prefix}/share/doc/${name} \
+                    --mandir=${prefix}/share/man \
+                    --with-includes=${prefix}/include \
+                    --with-libraries=${prefix}/lib \
+                    --with-openssl \
+                    --with-bonjour \
+                    --with-readline \
+                    --with-zlib \
+                    --with-lz4 \
+                    --with-zstd \
+                    --with-libxml \
+                    --with-libxslt \
+                    --with-gssapi \
+                    --enable-thread-safety \
+                    --with-uuid=e2fs \
+                    --with-icu
+
+configure.ldflags-append    -headerpad_max_install_names
+configure.universal_args-delete --disable-dependency-tracking
+
+# building psql with clang from Xcode prior to 4.4 causes segfault on query; see #31717
+compiler.blacklist-append {clang < 421}
+compilers.choose          cc cxx
+compilers.setup           -gcc -fortran -clang15 -clang33 -clang34 -clang37
+notes "To use the postgresql server as a startup item install the ${name}-server port"
+
+if {![info exists universal_possible]} {
+    set universal_possible [expr {${os.universal_supported} && [llength ${configure.universal_archs}] >= 2}]
+}
```

Comment:
This is ancient compatibility code for old versions of MacPorts base that don't have `universal_possible`.